### PR TITLE
3.0: Session::clear()

### PR DESCRIPTION
**Pull request by dereuromark**
Currently
- Session::check($name = null)
- Session::read($name = null)

work with `$name = null`, as well, reading all the session content.

I propose to also allow `Session::consume()` and `Session::delete()` to do that.
This would allow using those methods to work on the whole session (e.g. for setUp emptying) instead of having to workaround it using `clear()` or even `destroy()` where not even necessary, and no new session id has to be generated.

So this allows for a more consistent API.

Since only strict null allows that to happen, using falsely values like empty strings have no effect (allowing this to be BC as well) which should be the case IMO.

Also fixed a doc block statement.

### CODE REVIEW

File: tests/TestCase/Network/SessionTest.php [hunk 1]
```diff
@@ -321,6 +344,29 @@ public function testDelete()
...
+     * @return void
+     */
+    public function testDeleteAll()
+    {
+        $session = new Session();
+        $session->write('Some.string', 'value');
+        $session->write('Some.array', ['key1' => 'value1', 'key2' => 'value2']);
+
+        $session->delete();
+        $expected = [
```
Comment (antograssiot) [inline tests/TestCase/Network/SessionTest.php]: $expected is not needed

Comment (dereuromark) [inline tests/TestCase/Network/SessionTest.php]: bad copy paste :) thx.

File: src/Network/Session.php [hunk 2]
```diff
@@ -396,14 +396,19 @@ public function read($name = null)
...
+     * @param string|null $name The key to read and remove (or a path as sent to Hash.extract).
      * @return mixed The value of the session variable, null if session not available,
      *   session not started, or provided name not found in the session.
      */
-    public function consume($name)
+    public function consume($name = null)
     {
+        if ($name === null) {
+            $session = $_SESSION;
+            $_SESSION = null;
```
Comment (markstory) [inline src/Network/Session.php]: Shouldn't it become an array?

**Comment by lorenzo** [comment]
Have you tried this in browser? I think setting the $_SESSION variable to null does not delete all values. At least that is how PHP used to work

**Comment by dereuromark** [comment]
@lorenzo I think @markstory was right, it should reset to empty array in order for _overwrite() to function properly.

**Comment by lorenzo** [comment]
Ok, but did you try it in a browser?

**Comment by dereuromark** [comment]
@lorenzo Yes, I used a basic cake3 app to verify the behavior in a controller action.
And it works as expected there, as well.

**Comment by dereuromark** [comment]
What exactly is the difference of clear() and destroy() in regards of user functionality?
Because, thinking about consistency and https://github.com/cakephp/cakephp/pull/5545 it would make sense to have clear() here actually do what delete(null) does, and have clear() named something else, like renew(). That would make it more consistent in total IMO.

**Comment by dereuromark** [comment]
OK, as discussed it would be better to have clear() with `$reset = false (/true)` and only conditionally renew the session. And keep the current functionality.

**Comment by markstory** [comment]
I agree that having a separate clear() method would better communicate the intent of the method.

**Comment by dereuromark** [comment]
Docs added, as well.

**Pull request merged**